app/stores/base/Store.ts

```diff
@@ -131,7 +128,11 @@ export default abstract class Store<T extends Model> {
           }
 
           if (deleteBehavior === "cascade") {
-            store.remove(item.id);
+            if (item instanceof ParanoidModel) {
+              item.deletedAt = new Date().toISOString();
+            } else {
+              store.remove(item.id);
+            }
```

Comment:
Regressed in https://github.com/outline/outline/pull/7460/files#diff-09465a4af35119447c885e5f289deca73e86112bb6574950d306f872ba4bb161L124-L128

Reply:
Yea it was removed because it caused a worse bug... I can't remember the exact scenario but basically it makes it impossible to actually remove these models from the store when you want to as they will always get marked as deleted instead